pkg/apis/monitoring/v1/types.go

```diff
@@ -474,6 +474,17 @@ type PrometheusSpec struct {
 	// Note that if TargetLimit is lower, that value will be taken instead,
 	// except if either value is zero, in which case the non-zero value will be
 	// used.  If both values are zero, no limit is enforced.
+
+	// Storage related settings that are runtime reloadable.
+	StorageSettings *StorageSettingsSpec `json:"storageSettingsSpec,omitempty"`
```

Comment:
I'm not sure that we should replicate the Prometheus configuration format. We can probably avoid the `storageSettingsSpec` key.

```
spec:
  examplars:
    maxSize: 10000
```

Reply:
I will rename it :)

Reply:
I miss read your last comment @simonpasquier , so you want it to be:

```yaml
PrometheusSpec:
    exemplars:
        max_exemplars: 10000
```

What if they add new settings in the storage? 

If this is a blocker I will update the PR right now :)